### PR TITLE
Feature/crop cloud

### PR DESCRIPTION
**Pull request by jrslagle**
# Description

Added two endpoints for generating a crop cloud.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [X] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

## Has This Been Tested

- [X] Yes
- [ ] No
- [ ] Not necessary

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

**Review by ike-okonkwo — DISMISSED** [review]

File: .gitignore [hunk 1]
```diff
@@ -118,6 +118,9 @@ ENV/
 env.bak/
 venv.bak/
 
+# Google Credentials
+app/tmp/google.json
+
```
Comment (TemsyChen) [inline .gitignore:123]: Nice addition

File: app/utils/visualizations/crop_cloud.py [hunk 2]
```diff
@@ -0,0 +1,659 @@
...
+            y1 = pick_y(canvas_height=canvas.shape[0], word_height=image.shape[0])
+            y2 = y1 + image.shape[0]
+
+            R, G, B, mask = cv2.split(image)  # split the image into channels
+            color = cv2.merge((B, G, R))
+            mask = np.atleast_3d(mask) / 255
+
+            mask_bool = mask.reshape(mask.shape[:2]).astype('bool')
+            intersection = np.logical_and(mask_bool, occupied[y1:y2, x1:x2])
+
```
Comment (peterger8y) [inline app/utils/visualizations/crop_cloud.py:442]: I've said it before, and I'll say it again: really impressed by the from scratch crop cloud code here. Im guessing Ill implement color changes somewhere around here....

File: app/utils/visualizations/crop_cloud.py [hunk 3]
```diff
@@ -0,0 +1,659 @@
...
+                date_number=0,
+                total_dates=1,
+                )
+            x2 = x1 + image.shape[1]
+
+            # pick a vertical position
+            y1 = pick_y(canvas_height=canvas.shape[0], word_height=image.shape[0])
+            y2 = y1 + image.shape[0]
+
+            R, G, B, mask = cv2.split(image)  # split the image into channels
```
Comment (peterger8y) [inline app/utils/visualizations/crop_cloud.py:436]: I wonder if the complete split is necessary.... maybe just a partial split? as the R G B are immediately re-merged.

**Review by peterger8y — APPROVED** [review]
Very well done. Looking forward to getting this fully plugged in and implemented. 
Way to go!!

**Review by afroman32 — APPROVED** [review]
Very well written and easy to follow

**Review by ike-okonkwo — APPROVED** [review]
Thanks for the update

**Pull request merged**